File: python/examples/error_bars.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, error_bar_series, save_example

META = ExampleMeta(
    slug="error-bars",
    title="Vertical error bars",
    summary="A line-like series with y-direction uncertainty.",
    section="Statistical plots",
)


def build_plot():
    x, y, errors = error_bar_series()
    return (
        base_plot("Vertical Error Bars")
        .xlabel("trial")
        .ylabel("measurement")
        .error_bars(x, y, errors)
    )


if __name__ == "__main__":
    save_example(META, build_plot())

File: packages/ruviz/examples/error-bars.ts
import { createPlot } from "../src/index.js";
import { errorBarSeries } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "error-bars",
  title: "Vertical error bars",
  summary: "A series with y-direction uncertainty.",
  category: "statistical",
  createPlot() {
    const { x, y, errors } = errorBarSeries();
    return createPlot()
      .sizePx(640, 320)
      .title("Vertical Error Bars")
      .xlabel("trial")
      .ylabel("measurement")
      .errorBars({ x, y, yErrors: errors });
  },
};
